plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -190,54 +262,49 @@ export class DeploymentManager {
     }
   }
 
-  /* Stores a new alias, which can then be referenced via `deploymentManager.contract()` */
-  async putAlias(alias: Alias, address: Address) {
-    await putAlias(this.cache, alias, address);
-    this.invalidateContractsCache();
-  }
-
-  /* Stores a new proxy, which dictates the ABI available for that contract in `deploymentManager.contract()` */
-  async putProxy(alias: Alias, address: Address) {
-    await putProxy(this.cache, alias, address);
-    this.invalidateContractsCache();
-  }
-
-  /* Stores new roots, which are the basis for spidering. */
-  async putRoots(roots: Roots) {
-    await putRoots(this.cache, roots);
-  }
-
-  /* Gets the existing roots. */
-  async getRoots(): Promise<Roots> {
-    return await getRoots(this.cache);
-  }
-
   /* Loads contract configuration by tracing from roots outwards, based on relationConfig. */
-  async spider() {
-    let relationConfigMap = getRelationConfig(
+  async spider(deployed: Deployed = {}): Promise<Spider> {
+    const relationConfigMap = getRelationConfig(
       this.hre.config.deploymentManager,
       this.network,
       this.deployment
     );
-    let roots = await getRoots(this.cache);
-    let { aliases, proxies } = await spider(
+    const roots = new Map([
+      ...await getRoots(this.cache),
+      ...Object.entries(deployed).map(([a, c]): [Alias, Address] => [a, c.address])
+    ]);
+    const crawl = await spider(
       this.cache,
       this.network,
       this.hre,
       relationConfigMap,
       roots
     );
-    await storeAliases(this.cache, aliases);
-    await storeProxies(this.cache, proxies);
+    await putRoots(this.cache, roots);
+    await storeAliases(this.cache, crawl.aliases);
+    await storeProxies(this.cache, crawl.proxies);
+    this.invalidateContractsCache(); // TODO should we just write new contracts from spider?
+    return crawl;
+  }
+
+  /* Stores a new alias, which can then be referenced via `deploymentManager.contract()` */
+  async putAlias(alias: Alias, address: Address) {
+    await putAlias(this.cache, alias, address);
+    this.invalidateContractsCache();
```

Comment:
I forget why we need to do this, but I think it's meant to basically try and save some in-memory space by keeping some things cached so we don't overcall `new Contract`

Reply:
Which is, `putAlias`? I just moved this one down, if you are talking about the fact that it invalidates? We definitely have to rebuild contracts after this...which agree we could optimize all these invalidations but can def follow that one on

Reply:
Yeah, I'm just arguing that we could do better than invalidating here every time.